src/RequestForm.js

```diff
@@ -318,55 +302,87 @@ class RequestForm extends React.Component {
-          requestPreferencesLoaded: true,
-        });
-      })
-      .catch(() => {
-        this.setState({ requestPreferencesLoaded: true });
-      });
+  getDefaultRequestPreferences() {
+    return {
+      hasDelivery: false,
+      requestPreferencesLoaded: false,
+      defaultDeliveryAddressTypeId: '',
+      defaultServicePointId: '',
+      selectedDelivery: isDelivery(this.props.initialValues),
+      selectedAddressTypeId: this.props.initialValues.deliveryAddressTypeId || '',
+      fulfillmentPreference: fulfilmentTypeMap.HOLD_SHELF,
+    };
   }
 
-  setDefaultRequestPreferences = () => {
+  async findRequestPreferences(userId) {
     const {
-      hasDelivery,
-      fulfillmentPreference,
-      defaultServicePointId,
-      defaultDeliveryAddressTypeId,
-    } = this.state;
+      findResource,
+      change,
+    } = this.props;
+
+    try {
+      const { requestPreferences } = await findResource('requestPreferences', userId, 'userId');
+      const preferences = requestPreferences[0];
+
+      const requestPreference = {
+        ...this.getDefaultRequestPreferences(),
+        ...pick(preferences, ['defaultDeliveryAddressTypeId', 'defaultServicePointId']),
+        requestPreferencesLoaded: true,
+      };
 
-    this.props.change('fulfilmentPreference', fulfillmentPreference);
+      const deliveryIsPredefined = get(preferences, 'delivery');
+
+      if (isBoolean(deliveryIsPredefined)) {
+        requestPreference.hasDelivery = isDelivery;
+      }
+
+      const fulfillmentPreference = get(preferences, 'fulfillment');
+      const deliverySelected = this.isDeliverySelected(fulfillmentPreference);
 
-    if (fulfillmentPreference === fulfilmentTypeMap.DELIVERY) {
       this.setState({
-        selectedDelivery: true,
-        selectedAddressTypeId: defaultDeliveryAddressTypeId,
+        ...requestPreference,
+        selectedDelivery: deliverySelected,
+      }, () => {
+        change('fulfilmentPreference', fulfillmentPreference);
+
+        this.updateRequestPreferencesFields(fulfillmentPreference);
+      });
+    } catch (e) {
+      this.setState({
+        ...this.getDefaultRequestPreferences(),
+        requestPreferencesLoaded: true,
+        selectedDelivery: false,
+      }, () => {
+        change('fulfilmentPreference', fulfilmentTypeMap.HOLD_SHELF);
+        change('deliveryAddressTypeId', '');
       });
-      this.props.change('deliveryAddressTypeId', defaultDeliveryAddressTypeId);
     }
-    if (fulfillmentPreference === fulfilmentTypeMap.HOLD_SHELF) {
-      this.setState({ selectedDelivery: false });
-      this.props.change('pickupServicePointId', defaultServicePointId);
+  }
+
+  updateRequestPreferencesFields(fullfillmentPreference) {
+    const {
+      defaultDeliveryAddressTypeId,
+      defaultServicePointId,
+    } = this.state;
+    const { change } = this.props;
+
+    if (fullfillmentPreference === fulfilmentTypeMap.DELIVERY) {
```

Comment:
`state.selectedDelivery` can be used here making `fullfillmentPreference` parameter unnecessary